Khan/vite
# Close socket on unload

When proxying websockets using viteproxy, client sockets don't get disposed of properly when reloading the page.  This results in the following error in the vitejs-dev-server:
```
RangeError: Invalid WebSocket frame: invalid status code 1006
    at Receiver$1.controlMessage (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:50513:18)
    at Receiver$1.getData (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:50381:42)
    at Receiver$1.startLoop (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:50106:22)
    at Receiver$1._write (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:50041:10)
    at doWrite (_stream_writable.js:403:12)
    at writeOrBuffer (_stream_writable.js:387:5)
    at Receiver$1.Writable.write (_stream_writable.js:318:11)
    at Socket.socketOnData (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:52455:37)
    at Socket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:297:12)
Emitted 'error' event on WebSocket$2 instance at:
    at Receiver$1.receiverOnError (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:52342:13)
    at Receiver$1.emit (events.js:314:20)
    at errorOrDestroy (internal/streams/destroy.js:108:12)
    at onwriteError (_stream_writable.js:418:5)
    at onwrite (_stream_writable.js:445:5)
    at Receiver$1.startLoop (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:50115:5)
    at Receiver$1._write (/Users/kevinbarabash/khan/webapp/services/static/node_modules/vite/dist/node/chunks/dep-713b45e1.js:50041:10)
    [... lines matching original stack trace ...]
    at Receiver$1.Writable.write (_stream_writable.js:318:11) {
  code: 'WS_ERR_INVALID_CLOSE_CODE',
  [Symbol(status-code)]: 1002
```

This PR fixes the issue by adding an `unload` event listener to the page which calls `socket.close()` .  This closes the socket normally and avoids the error.

## Test Plan
- apply this change manually to services/static/node_modules/vite/dist/client/client.mjs
- reload vitejs service in hotel (make sure vitejs-proxy is also running)
- load localhost:8088
- reload the page, see vitejs continues to run without any errors and that the page reloads as expected
